components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/CommentsManagerImpl.java

```diff
@@ -17,10 +17,274 @@
+        validateComment(apAppCommentId,comment.getComment());
+
+        if (log.isDebugEnabled()) {
+            log.debug("Comment retrieval request is received for the comment id " +apAppCommentId);
+        }
+        comment.setModifiedAt(Timestamp.from(Instant.now()));
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+            return ApplicationManagementDAOFactory.getCommentDAO().updateComment(apAppCommentId,
+                    comment.getComment(),comment.getModifiedBy(),comment.getModifiedAt());
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+    }
+
+    public int getRatedUser(String uuid){
+
+        int ratedUsers=0;
+        if (log.isDebugEnabled()) {
+            log.debug("Get the rated users for the application release number"+uuid);
+        }
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            ratedUsers =ApplicationManagementDAOFactory.getCommentDAO().getRatedUser(uuid);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (ApplicationManagementDAOException e) {
+            log.error("Application Management Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return ratedUsers;
+    }
+
+    @Override
+    public int getStars(String uuid) throws SQLException {
+
+        int stars=0;
+        if (log.isDebugEnabled()) {
+            log.debug("Get the average of rated stars for the application "+uuid);
+        }
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            stars= ApplicationManagementDAOFactory.getCommentDAO().getStars(uuid);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (ApplicationManagementDAOException e) {
+            log.error("Application Management Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return stars;
+    }
+
+    @Override
+    public int updateStars(int stars , String uuid) throws ApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Stars are received for the application " + uuid);
+        }
+        int newStars=0;
+        try {
+            ConnectionManagerUtil.beginDBTransaction();
+
+            int ratedUsers = ApplicationManagementDAOFactory.getCommentDAO().getRatedUser(uuid);
+            int oldStars = ApplicationManagementDAOFactory.getCommentDAO().getStars(uuid);
```

Comment:
Since we are accessing `CommenDAO` several times; `ApplicationManagementDAOFactory.getCommentDAO()` can be moved to a variable similar to `commentDAO`